Change supported tools list to align with PDD

jcollins-g opened the pull request:
This will break all clients (without the changes I have in progress locally in the SDK, will break it too).  So update major version number.

I am hoping that it is valid to have dashes in the tool name as the PDD specifies them with dashes, so added that to the regexp.

@keertip

eliasyishak commented on pkgs/unified_analytics/test/unified_analytics_test.dart:
```diff
@@ -1013,7 +1013,7 @@ $initialToolName=${ConfigHandler.dateStamp},$toolsMessageVersion
       // All tools must be under 36 characters (and enforce each tool
       // begins with a letter)
-      final RegExp toolLabelPattern = RegExp(r'^[a-zA-Z][a-zA-Z\_]{0,35}$');
+      final RegExp toolLabelPattern = RegExp(r'^[a-zA-Z][a-zA-Z\_-]{0,35}$');
```
There are some restrictions in Google Analytics about sending some parameters in the body with only alphanumeric and underscores so I think it would be best to apply that for all items where we can

https://developers.google.com/analytics/devguides/collection/protocol/ga4/sending-events?client_type=gtag#limitations

jcollins-g replied:
OK.  We will then have to have a separate field for the control flag and not use the same field for the control flag and the label.  I'll make that change.

jcollins-g replied:
Chat offline indicates that the value sent to analytics can contain dashes.  So keeping the control flag and label as the same thing is fine.

eliasyishak commented on pkgs/unified_analytics/lib/src/enums.dart:
```diff
@@ -70,21 +63,34 @@ enum DashEvent {
-  languageServer(
-    label: 'language_server',
-    description: 'The Dart language server for IDE and CLI support.',
+  vscodePlugins(
+    label: 'vscode-plugins',
+    description: 'VS Code IDE plugins for Dart and Flutter',
+  ),
+  intellijPlugins(
+    label: 'intellij-plugins',
+    description: 'IntelliJ IDE plugins for Dart and Flutter',
+  ),
+  androidStudioPlugins(
+    label: 'android-studio-plugins',
+    description: 'Android Studio IDE plugins for Dart and Flutter',
+  ),
+  devtools(
+    label: 'devtools',
+    description: 'DevTools debugging and performance tools',
```
Small nit but can we add these in alphabetical order?

jcollins-g replied:
I can.  I ordered them the order they are in in the PDD for my peace of mind, but fortunately that isn't a requirement :-D

jcollins-g commented on pkgs/unified_analytics/lib/src/enums.dart:
```diff
@@ -19,44 +19,37 @@ enum DashEvent {
   clientNotification(
     label: 'client_notification',
     description: 'Notifications sent from the client',
-    toolOwner: DashTool.languageServer,
```
Correct.  All logical tool groups that indirectly use the language server can send this.

jcollins-g reviewed (commented):
(not yet ready for re-review, will post explicitly when ready)

keertip commented on pkgs/unified_analytics/lib/src/enums.dart:
```diff
@@ -70,21 +63,34 @@ enum DashEvent {
+  dartTool(
+    label: 'dart-tool',
+    description: 'Dart CLI developer tool',
   ),
```
Can we leave the previous value in too and add the new one. Then once this lands in the SDK, we can move over the code to the new enum value and then remove the old ones. That way it is not a breaking change, and we will not break the SDK build when DEPS is updated to pull in the new package.

jcollins-g replied:
OK, I was thinking on the old model where DEPS were updated manually and I'd figure I'd land my changes on top of it, but I forgot there's now a semi-automatic update in place.   I can do this.

eliasyishak commented:
> OK. We will then have to have a separate field for the control flag and not use the same field for the control flag and the label. I'll make that change.

@jcollins-g is it necessary that we label only with hyphens as opposed to underscores? I only ask because having separate strings for what we store in the config file and what we send to GA can make it difficult to debug if functionality around checking the config file has bugs.

bwilkerson commented:
I'd like to get clarification from Michael about the tool names and whether then need to match the PDD. It would read a lot better in the notification if we didn't have either a hyphen or an underscore. For example, rather than 'dart-tool', could we say 'dart command-line tool'? Being more clear for our users would, imo, be better.

eliasyishak commented:
It wouldn't be a bad idea to add another parameter for the enhanced enum in addition to `label` and `description` that we can use to inject into the consent message as well.

jcollins-g commented:
Updated.  Also fixed a bug where the 'description' string wasn't being used in the consent message, which probably was part of the confusion between us around that.

bwilkerson commented:
Elias can correct me, but it was my understanding that the 'description' string was for documentation purposes, and not for use in the message.

eliasyishak commented:
That's correct @bwilkerson , we were not intending to use the description there as those can get pretty verbose. If we want to use the enhanced enum to inject the name into the consent message, we should add a separate variable outside of `label` and `description`

bwilkerson commented:
I can't remember the use case for having a description field (as opposed to having a comment). Unless the use case is still valid, maybe we should just remove the 'description' field.

jcollins-g commented:
The description strings (as I have them in this CL) however are explicitly required in the PDD.   I don't actually think it's that terrible looking and is better than what we had before, if you check the output:  https://github.com/dart-lang/tools/pull/55/files#diff-a60799c7bf90e870bc18c7ab4d1a0f885ccab63b0776c8ae10561885a57e25adR1102

eliasyishak commented:
I see, well in that case, we can just repurpose the `description` to match what the PDD has them described as and just inject them into the consent string. I think it would be safe to assume that anyone contributing to these tools would understand from which tool is which based on the PDD values from the table

jcollins-g commented:
> 

I added a comment indicating that the `description` is actually the "notice string" from that table.  Hopefully that's clear.

jcollins-g commented:
Updated from head.  Please take a look.

eliasyishak reviewed (approved):
LGTM
